browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5671,6 +5687,35 @@ L.CanvasTileLayer = L.Layer.extend({
 		}
 	},
 
+	_onWorkerMessage: function(e) {
+		switch (e.data.message) {
+		case 'decompress':
+			var params = this._workerMap.get(e.data.tile.id);
+			this._workerMap.delete(e.data.tile.id);
+			if (params.isKeyframe)
+				params.tile.hasPendingKeyframe = false;
+			else
+				params.tile.hasPendingDelta = false;
+			this._applyDelta(params.tile, e.data.tile.rawDelta, e.data.tile.deltas, params.isKeyframe, params.wireMessage, false);
+			this._workerBuffers.push(e.data.buffer);
+
+			for (var i = 0; this._workerBuffers.length && i < this._workerPendingEvents.length;) {
+				var params = this._workerPendingEvents[i];
+				if (!params.tile.hasPendingUpdate()) {
+					this._workerPendingEvents.splice(i, 1);
+					this._applyCompressedDelta(params.tile, params.rawDelta, params.isKeyframe, params.wireMessage);
+				} else
+					++i;
+			}
```

Comment:
This slightly convoluted thing is to deal with the possibility of a tile getting a second update before the first is handled (e.g. a keyframe and a delta in quick succession) - it's incredibly unlikely, but I did hit it while testing.